lib/Transforms/Scalar/PromoteVerifierCalls.cc

```diff
@@ -173,6 +174,18 @@ bool PromoteVerifierCalls::runOnFunction(Function &F) {
       if (cg)
         (*cg)[&F]->addCalledFunction(ci, (*cg)[ci->getCalledFunction()]);
 
+      I.replaceAllUsesWith(ci);
+      toKill.push_back(&I);
+    } else if (fn && (fn->getName().equals("sea_assert_if"))) {  // sea_assert_if is the user facing name
+      IRBuilder<> Builder(F.getContext());
+      Builder.SetInsertPoint(&I);
+      // arg0: antecedent
+      // arg1: consequent
+      CallInst *ci =
+          Builder.CreateCall(m_assert_if, {CS.getArgument(0), CS.getArgument(1)});
```

Comment:
the arguments are unlikely to be `i1`. They would be a `zext` of `i1` to something expected by the function. See how `verifier.assume` is lifted.

Reply:
done